packages/maps/src/components/basic/GeoDistanceSlider.js

```diff
@@ -52,71 +45,68 @@ class GeoDistanceSlider extends Component {
 		if (props.autoLocation) {
 			this.getUserLocation();
 		}
-		props.setQueryListener(props.componentId, props.onQueryChange, null);
-	}
 
-	componentWillMount() {
-		this.props.addComponent(this.props.componentId);
-		this.setReact(this.props);
+		let currentLocation = null;
 
-		if (this.props.selectedValue && this.props.selectedValue.location) {
-			this.setState(
-				{
-					currentLocation: this.props.selectedValue.location,
-				},
-				() => {
-					this.getCoordinates(this.props.selectedValue.location, () => {
-						if (this.props.selectedValue.distance) {
-							this.setDistance(this.props.selectedValue.distance);
-						}
-					});
-				},
-			);
-		} else if (this.props.defaultSelected && this.props.defaultSelected.location) {
-			this.setState(
-				{
-					currentLocation: this.props.defaultSelected.location,
-				},
-				() => {
-					this.getCoordinates(this.props.defaultSelected.location, () => {
-						if (this.props.defaultSelected.distance) {
-							this.setDistance(this.props.defaultSelected.distance);
-						}
-					});
-				},
-			);
+		if (props.selectedValue && props.selectedValue.location) {
+			currentLocation = props.selectedValue.location;
+			this.getCoordinates(props.selectedValue.location, () => {
+				if (props.selectedValue.distance) {
+					this.setDistance(props.selectedValue.distance);
+				}
+			});
+		} else if (props.defaultSelected && props.defaultSelected.location) {
```

Comment:
we should remove `defaultSelected` and use `value` - `defaultValue` props.

Reply:
Have done defaultValue changes but wasnt able to make it control.